certbot/certbot/_internal/display/obj.py

```diff
@@ -532,16 +530,11 @@ def directory_select(self, message, default=None,
         return self.input(message, default, cli_flag)
 
 
-# The two following functions use "Any" for their parameter/output types. Normally interfaces from
-# certbot.interfaces would be used, but MyPy will not understand their semantic. These interfaces
-# will be removed soon and replaced by ABC classes that will be used also here for type checking.
-# TODO: replace Any by actual ABC classes once available
```

Comment:
I'm not planning on blocking this PR on it, but I wonder if we can use a union of the two display types here?

If that doesn't work for some reason, it's probably not worth worrying about it now, but if we wanted to do something else here, a few ideas I have are:

1. Define an ABC for this interface, but keep it under `_internal`.
2. Do further refactoring on how all the plumbing under `certbot.display.util` is structured. For instance, we could replace these heavyweight classes with simple structures maintaining things like where output should be written and those structured could either be used directly by the code in `certbot.display.util` or in helper functions here.

Reply:
It works perfectly fine with an Union. I added that.